controllers/machinedeletionremediation_controller.go

```diff
@@ -221,77 +210,100 @@ func (r *MachineDeletionRemediationReconciler) buildMachineFromNode(node *v1.Nod
-	annotations := remediation.GetAnnotations()
-	if annotations == nil {
-		annotations = make(map[string]string, 1)
+func getMachineNameNsFromNode(node *v1.Node) (string, string, error) {
+	var nodeAnnotations map[string]string
+	if nodeAnnotations = node.Annotations; nodeAnnotations == nil {
+		return "", "", fmt.Errorf(noAnnotationsError, node.Name)
 	}
 
-	annotations[MachineNameNamespaceAnnotation] = fmt.Sprintf("%s/%s", machine.GetNamespace(), machine.GetName())
-	remediation.SetAnnotations(annotations)
+	var machineNameNs string
+	if machineNameNs = nodeAnnotations[machineAnnotationOpenshift]; len(machineNameNs) == 0 {
+		return "", "", fmt.Errorf(noMachineAnnotationError, node.Name)
+	}
 
-	return r.Update(ctx, remediation)
+	if slice := strings.Split(machineNameNs, "/"); len(slice) == 2 {
+		return slice[1], slice[0], nil
+	}
+	return "", "", fmt.Errorf(invalidValueMachineAnnotationError, node.Name)
 }
 
-func extractNameAndNamespace(nameNamespace string, nodeName string) (string, string, error) {
-	if nameNamespaceSlice := strings.Split(nameNamespace, "/"); len(nameNamespaceSlice) == 2 {
-		return nameNamespaceSlice[1], nameNamespaceSlice[0], nil
+func extractNameAndNamespace(nameNs string, nodeName string) (string, string, error) {
+	if nameNsSlice := strings.Split(nameNs, "/"); len(nameNsSlice) == 2 {
+		return nameNsSlice[1], nameNsSlice[0], nil
 	}
 	return "", "", fmt.Errorf(invalidValueMachineAnnotationError, nodeName)
 }
 
-func getMachineStatusPhase(machine *unstructured.Unstructured) (string, error) {
-	status, ok, err := unstructured.NestedMap(machine.Object, "status")
+func getMachineStatusPhase(machine *unstructured.Unstructured) string {
+	status, exists, err := unstructured.NestedMap(machine.Object, "status")
 	if err != nil {
-		return "", fmt.Errorf("could not get Machine's status: error %w", err)
+		ctrl.Log.WithName("getMachineStatusPhase").Error(err, "could not get machine's status", "machine", machine.GetName())
+		return "unknown"
 	}
-	if !ok {
-		return "", fmt.Errorf("Machine object does not have a status field")
+	if !exists {
+		ctrl.Log.WithName("getMachineStatusPhase").Info("machine does not have status", "machine", machine.GetName())
+		return "unknown"
 	}
 
-	phase, ok, err := unstructured.NestedString(status, "phase")
+	phase, exists, err := unstructured.NestedString(status, "phase")
 	if err != nil {
-		return "", fmt.Errorf("could not get Machine's status.phase: error %w", err)
+		ctrl.Log.WithName("getMachineStatusPhase").Error(err, "could not get machine's status", "machine", machine.GetName())
+		return "unknown"
 	}
-	if !ok {
-		return "", fmt.Errorf("Machine object does not have a status.phase field")
+	if !exists {
+		ctrl.Log.WithName("getMachineStatusPhase").Info("machine does not have phase", "machine", machine.GetName())
```

Comment:
How about setting `ctrl.Log.WithName("getMachineStatusPhase")` once as a global var in the beginning of the file, instead of doing that multiple times in *getMachineStatusPhase* function.

machineStatusLogger := ctrl.Log.WithName("getMachineStatusPhase")

And here just use 
```suggestion
		machineStatusLogger.Info("machine does not have phase", "machine", machine.GetName())
```

Reply:
Do you think it would be the same to declare it locally within the function? Generally, I prefer to keep things within a smaller context whenever possible.

Reply:
> Do you think it would be the same to declare it locally within the function?

Yes, give it a try. I didn't notice that you don't use this log in other functions.. Declaring locally makes perfect sense